sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ShouldRetryResult.java

```diff
@@ -17,41 +17,48 @@
     public final Exception exception;
     public final Quadruple<Boolean, Boolean, Duration, Integer> policyArg;
     public boolean shouldRetry;
+    public boolean nonRelatedException;
 
     private ShouldRetryResult(Duration dur, Exception e, boolean shouldRetry,
-                              Quadruple<Boolean, Boolean, Duration, Integer> policyArg) {
+                              Quadruple<Boolean, Boolean, Duration, Integer> policyArg, boolean nonRelatedException) {
         this.backOffTime = dur;
         this.exception = e;
         this.shouldRetry = shouldRetry;
         this.policyArg = policyArg;
+        this.nonRelatedException = nonRelatedException;
     }
 
     public static ShouldRetryResult error(Exception e) {
         Utils.checkNotNullOrThrow(e, "exception", "cannot be null");
-        return new ShouldRetryResult(null, e, false, null);
+        return new ShouldRetryResult(null, e, false, null, false);
     }
 
     public static ShouldRetryResult noRetry() {
-        return new ShouldRetryResult(null, null, false, null);
+        return new ShouldRetryResult(null, null, false, null, false);
+    }
+
+    public static ShouldRetryResult noRetryOnNonRelatedException() {
```

Comment:
just a suggestion: Instead of NonRelated, NonCosmos might be better for readability?

Reply:
NonCosmos  is not valid , example for GoneAndRetryPolicy 429 is not related , but it is still cosmos